lib/galaxy/model/migrations/util.py

```diff
@@ -1,81 +1,368 @@
 
+    def _log_object_exists_message(self, object_name: str) -> None:
+        log.info(f"{object_name} already exists. Skipping revision.")
 
-def create_index(index_name, table_name, columns):
-    if index_exists(index_name, table_name):
-        msg = f"Index with name {index_name} on {table_name} already exists. Skipping revision."
-        log.info(msg)
-    else:
-        op.create_index(index_name, table_name, columns)
+    def _log_object_does_not_exist_message(self, object_name: str) -> None:
+        log.info(f"{object_name} does not exist. Skipping revision.")
+
+
+class DDLAlterOperation(DDLOperation):
+    """
+    Base class for DDL operations that implement special handling of ALTER statements.
+    Ref:
+    - https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.batch_alter_table
+    - https://alembic.sqlalchemy.org/en/latest/batch.html
+    """
+
+    def __init__(self, table_name: str) -> None:
+        self.table_name = table_name
+
+    def run(self) -> Optional[Any]:
+        if context.is_offline_mode():
+            log.info("Generation of `alter` statements is disabled in offline mode.")
+            return None
+        return super().run()
+
+    def execute(self) -> Optional[Any]:
+        if _is_sqlite():
+            with legacy_alter_table(), op.batch_alter_table(self.table_name) as batch_op:
+                return self.batch_execute(batch_op)
+        else:
+            return self.non_batch_execute()  # use regular op context for non-sqlite db
+
+    @abstractmethod
+    def batch_execute(self, batch_op) -> Optional[Any]:
+        ...
+
+    @abstractmethod
+    def non_batch_execute(self) -> Optional[Any]:
+        ...
+
+
+class CreateTable(DDLOperation):
+    """Wraps alembic's create_table directive."""
+
+    def __init__(self, table_name: str, *columns: sa.schema.SchemaItem) -> None:
+        self.table_name = table_name
+        self.columns = columns
+
+    def execute(self) -> Optional[sa.Table]:
+        return op.create_table(self.table_name, *self.columns)
+
+    def pre_execute_check(self) -> bool:
+        return not table_exists(self.table_name, False)
+
+    def log_check_not_passed(self) -> None:
+        self._log_object_exists_message(f"{self.table_name} table")
+
+
+class DropTable(DDLOperation):
+    """Wraps alembic's drop_table directive."""
+
+    def __init__(self, table_name: str) -> None:
+        self.table_name = table_name
+
+    def execute(self) -> None:
+        op.drop_table(self.table_name)
+
+    def pre_execute_check(self) -> bool:
+        return table_exists(self.table_name, False)
+
+    def log_check_not_passed(self) -> None:
+        self._log_object_does_not_exist_message(f"{self.table_name} table")
+
+
+class CreateIndex(DDLOperation):
+    """Wraps alembic's create_index directive."""
+
+    def __init__(self, index_name: str, table_name: str, columns: Sequence, **kw: Any) -> None:
```

Comment:
Unfortunately this will fail. The `kw` value is passed to Alembic's [`op.create_index`](https://github.com/sqlalchemy/alembic/blob/main/alembic/operations/ops.py#L925), where the type is `Any`. Here's the resulting error:
```
`lib/galaxy/model/migrations/util.py:137: error: Argument 4 to "create_index" has incompatible type "**Dict[str, Dict[str, Any]]"; expected "bool"  [arg-type]
            op.create_index(self.index_name, self.table_name, self.columns, **self.kw)
``

Reply:
alright, thanks for checking.